soaktest/soaktest.go

```diff
@@ -0,0 +1,285 @@
+// monitors broker avg and p99 using datadog
+func rttStats(raw map[string]interface{}, topic string) error {
+	for _, broker := range raw["brokers"].(map[string]interface{}) {
+		var parts string
+		var avg float64
+		var p99 float64
+		var nodeid int
+
+		topparsValue := broker.(map[string]interface{})["toppars"]
+		if topparsValue == nil ||
+			len(topparsValue.(map[string]interface{})) == 0 {
+			continue
+		}
+
+		for _, val := range topparsValue.(map[string]interface{}) {
+			if val.(map[string]interface{})["topic"] != topic {
+				continue
+			}
+			partition := val.(map[string]interface{})["partition"]
+			f, ok := partition.(float64)
+			if !ok {
+				err := fmt.Errorf("Failed to convert partition "+
+					"to float64 %v\n", ok)
+				return err
+			}
+			if len(parts) > 0 {
+				parts += ","
+			}
+			parts += fmt.Sprintf("%d", int(f))
+		}
+
+		rttValue := broker.(map[string]interface{})["rtt"]
+		avg, ok := rttValue.(map[string]interface{})["avg"].(float64)
+		if !ok {
+			err := fmt.Errorf("failed to convert avg to float64 %v", ok)
+			return err
+		}
+
+		p99, ok = rttValue.(map[string]interface{})["p99"].(float64)
+		if !ok {
+			err := fmt.Errorf("failed to convert p99 to float64 %v", ok)
+			return err
+		}
+
+		nid, ok := broker.(map[string]interface{})["nodeid"].(float64)
+		if !ok {
+			err := fmt.Errorf("failed to convert nodeid to float64 %v", ok)
+			return err
+		}
+		nodeid = int(nid)
+
+		tags := []string{fmt.Sprintf("broker:%d, type:%s, topic:%s, parts:%s",
+			nodeid, raw["type"], topic, parts)}
+
+		DatadogGauge(brokerRttP99, p99/1000000.0, tags)
+		DatadogGauge(brokerRttAvg, avg/1000000.0, tags)
+	}
+	return nil
+}
+
+// HandleStatsEvent converts Stats events to hashmap
+func HandleStatsEvent(e *kafka.Stats, topic string) error {
+	var raw map[string]interface{}
+	err := json.Unmarshal([]byte(e.String()), &raw)
+	if err != nil {
+		ErrorLogger.Printf("Json unmarshall error: %s\n", err)
+		return err
+	}
+
+	err = rttStats(raw, topic)
+	if err != nil {
+		ErrorLogger.Printf("Failed to calculate broker rtt statistics: %s",
+			err)
+		return err
+	}
+	return nil
+}
+
+// ProducerDeliveryCheck handles delivery report for producer
+func ProducerDeliveryCheck(e *kafka.Message) {
+	tags := []string{fmt.Sprintf("topic:%s, partition:%d",
+		*e.TopicPartition.Topic, e.TopicPartition.Partition)}
+	if e.TopicPartition.Error != nil {
+		drErrCnt++
+		DatadogIncrement(producerDrErr, 1, tags)
+		ErrorLogger.Printf("Delivery failed: %v\n", e.TopicPartition)
+	} else {
+		drCnt++
+		DatadogIncrement(producerDr, 1, tags)
+		if drCnt%1000 == 0 {
+			InfoLogger.Printf("Delivered message to topic at events channel "+
+				"%s [%d] at offset %d\n", *e.TopicPartition.Topic,
+				e.TopicPartition.Partition, e.TopicPartition.Offset)
+		}
+	}
+}
+
+// PrintConsumerStatus prints the information for the consumer
+func PrintConsumerStatus() {
+	InfoLogger.Printf("Consumer: %d messages consumed, %d duplicates, "+
```

Comment:
Since there are two consumers, can we prefix this with either TxnConsumer or OutputConsumer?